community/modules/scripts/htcondor-install/files/autoscaler.py

```diff
@@ -260,45 +259,59 @@ def scale(self):
         currentTarget = int(responseGroupInfo["targetSize"])
         print("Current MIG target size: " + str(currentTarget))
 
+        currentActions= responseGroupInfo["currentActions"]
+        n_being_born = currentActions["creating"] + \
+            currentActions["recreating"] + \
+            currentActions["starting"] + \
+            currentActions["restarting"] + \
+            currentActions["verifying"]
+
+        # count nodes that are either idle and have joined the pool or are
+        # still joining the pool from the MIG; there is some potential for
+        # overlap with the "verifying" state that will be addressed by
+        # only deleting nodes whose uptime exceeds this period of overlap
+        n_idle = len(idle_node_ads)
+        n_idle_total = n_being_born + n_idle
+        print(f"Idle nodes found: {n_being_born} birthing and {n_idle} idling")
+
+        # always keep size tending toward the minimum idle VMs requested
+        delta = min_hosts_for_queue - currentTarget
+        if delta < self.compute_instance_min_idle:
+            min_hosts_for_queue += self.compute_instance_min_idle - delta - n_idle_total
```

Comment:
This block (L278,279,280) feels is quite confusing. 

I don't understand why we have the **if** statement. What if delta=10 and compute_instance_min_idle=2. I don't understand why in this case we wouldn't want to bump up to account for min idle.

It seems to me like we could just set new_target = min_hosts_for_queue + compute_instance_min_idle. This could be done without regard to what is currently idle.

Reply:
I am going to resolve this conversation while quoting it in the other conversation in same set of lines.